src/working_files.cc

```diff
@@ -327,20 +327,27 @@ void WorkingFiles::OnChange(const Ipc_TextDocumentDidChange::Params& change) {
   for (const Ipc_TextDocumentDidChange::lsTextDocumentContentChangeEvent& diff :
        change.contentChanges) {
     // std::cerr << "|" << file->buffer_content << "|" << std::endl;
-    // If range or rangeLength are emitted we replace everything, per the spec.
-    if (diff.rangeLength == -1) {
+    // Per the spec replace everything if the rangeLength and range are not set.
+    // See https://github.com/Microsoft/language-server-protocol/issues/9.
+    auto zeroPosition = lsPosition(0, 0);
```

Comment:
Prefer default ctor, avoid auto, use const. Consider moving `kZeroPosition` into `lsPosition` itself. 

```
constexpr const lsPosition kZeroPosition;
```

Reply:
when I add constexpr the compiler informs me that only literals can be constexpr, so it is just const kZeroPosition